docs/rules/no-hooks-from-ancestor-modules.md

```diff
@@ -0,0 +1,57 @@
+# Forbid the use of hooks from ancestor modules (no-hooks-from-ancestor-modules)
+
+When a QUnit `module` is used with a nested callback, the callback provides a `hooks`
+object as its first argument. This allows calling `hooks.beforeEach` and `hooks.afterEach`
+within that callback's body.
+
+More deeply nested `module` uses should use their own callback's provided `hooks` argument
+and not one from an ancestor `module`.
+
+## Rule Details
+
+The following patterns are considered warnings:
+
+```js
+
+QUnit.module("outer module", function(hooks) {
+  hooks.beforeEach(function() {});
+
+  QUnit.module("inner module", function() {
+    hooks.beforeEach(function() {});
+  });
+});
+
+
+QUnit.module("outer module", function(outerModuleHooks) {
+  outerModuleHooks.beforeEach(function() {});
+
+  QUnit.module("inner module", function(innerModuleHooks) {
+    innerModuleHooks.beforeEach(function() {});
+    outerModuleHooks.beforeEach(function() {});
+  });
+});
+
+```
+
+The following patterns are not warnings:
+
+```js
+
+QUnit.module("outer module", function(hooks) {
+  hooks.beforeEach(function() {});
```

Comment:
This line should not warn regardless, so it might be good to simplify this example by removing this line (or maybe just adding a comment saying no warning would appear on this line). That way, the reader can focus on the inner `hooks.beforeEach` and how it will also not warn.

Reply:
The diff is behaving weirdly for me here-- would you mind summarizing what changes you made in this area (if any)? Sorry for having lost the context here.

Reply:
In the [first iteration](https://github.com/platinumazure/eslint-plugin-qunit/commit/3889d07fd2e2d165fb8cad977c8022e3efb3df7f) I had another nested `QUnit.module` invocation with its own `hooks.beforeEach` call inside this one, which I have since removed. Would it be clearer keeping the nesting but just drop the top level hook?